### ISSUE TITLE
Errors while setting up a run

### ISSUE THREAD

**Issue by gyebra-phs** [CDCgov/MIRA-NF#28]
### Description of the bug

Hi all,

First of all, thanks for making this tool available. I'm trying to test it with a few influenza A Illumina reads, but I'm afraid I'm encountering some errors early in the process that seem to show the pipeline can't find the reads.

Just to make sure I'm following the instructions, I have created a run directory (`runpath`) which contains a `fastqs` directory with all the fastq files, a sample sheet as described in the instructions (see below) and the primers used in fasta format.

```
head <runpath>/samplesheet.csv
Sample ID,Sample Type
Sample1,Test
Sample2,Test
Sample3,Test
```
(e.g. the fastq files for Sample1 follow the naming system `<runpath>/fastqs/Sample1_R1_001.fastq.gz` and `<runpath>/fastqs/Sample1_R2_001.fastq.gz`)

If I try to run the pipeline as:

```
nextflow run ./main.nf \
   -profile singularity,local \
   --input "$runpath"/samplesheet.csv \
   --outdir "$runpath"/outputs \
   --runpath "$runpath" \
   --e Flu-Illumina \
   --custom_primers "$runpath"/primers.fasta \
   --read_qc true
```

I get the following error:

```
Command error:
  Traceback (most recent call last):
    File "/data/tools/MIRA-NF/bin/create_nextflow_samplesheet_i.py", line 53, in <module>
  argv[1:]= ['-s', 'samplesheet.csv', '-r', '<runpath>/outputs', '-e', 'Flu-Illumina']
  <runpath>/outputs
  samplesheet.csv
      R1_fastq = glob(f"{runpath}/fastqs/{id}*R1*fastq*", recursive=True)[0]
  IndexError: list index out of range
```

I see that the pipeline creates another `fastqs` folder within the outdir, is this intended?

I would appreciate any help, thanks!

Gonzalo

**Comment by gyebra-phs** [comment]
Hi Kristine,

Thanks for your quick reply, my samples have names very similar to things like `FLUA_Run030_Pooled_PC_1_S370_L001_R1_001.fastq.gz`.

Thanks!

**Comment by kristinelacek** [comment]
Thanks so much! So does your samplesheet say "Sample1", "Sample2"? Your samplesheet has to match part of the fastq filenames you provide for illumina data.

For example, your samplesheet would look like

Sample ID,Sample Type
FLUA_Run030_Pooled_PC_1,Test
FLUA_Run030_Pooled_PC_2,Test

MIRA is looking for those sample IDs in the fastq file names

**Comment by gyebra-phs** [comment]
Sorry if I wasn't clear! My samplesheet looks like:

```
Sample ID,Sample Type
FLUA_Run030_Pooled_PC_1_S370_L001,Test
FLUA_Run030_Pooled_PC_2_S371_L001,Test
```

The corresponding files would be:

```
 <runpath>/fastqs/FLUA_Run030_Pooled_PC_1_S370_L001_R1_001.fastq.gz
 <runpath>/fastqs/FLUA_Run030_Pooled_PC_1_S370_L001_R2_001.fastq.gz
 <runpath>/fastqs/FLUA_Run030_Pooled_PC_2_S371_L001_R1_001.fastq.gz
 <runpath>/fastqs/FLUA_Run030_Pooled_PC_2_S371_L001_R2_001.fastq.gz
```

Do you think the extra "_001" after R1/R2 might be causing this issue? I saw in the code that it used wildcards (`R1_fastq = glob(f"{runpath}/fastqs/{id}*R1*fastq*", recursive=True)[0]`) so I thought they would be ok.

**Comment by mandysulli** [comment]
Hello Gonzalo,

Sometimes this error occurs when there is an empty line at the end of the samplesheet. Could you open the samplesheet in your linux/unix environment and be sure that there are no empty lines at the end?

Mandy

**Comment by gyebra-phs** [comment]
Hi Mandy,

I'm afraid there's no empty lines at the end of my sample sheet!

**Comment by kristinelacek** [comment]
Are you using an IDE?  I have found, at times, vim will show no extra line but `code samplesheet.csv` in VSCode does! Worth a check, just in case

**Comment by mandysulli** [comment]
Other things that can case this are duplicate names within the samplesheet.csv or if there is a missing read 1 or read 2 files for a sample. Could you check these things as well?

**Comment by gyebra-phs** [comment]
Thanks both for your suggestions, I'm generating the sample sheet from the command line (precisely to avoid having extra hidden characters!) like this:

```
cd <runpath>/fastqs
echo "Sample ID,Sample Type" > ../samplesheet.csv
ls *R1_001.fastq.gz | sed 's/_R1_001\.fastq\.gz/,Test/g' >> ../samplesheet.csv
```
So I don't think it's due to those issues?

On the other hand, I'm trying to understand what the pipeline does, specifically the first step (the staging process): is it copying the files in `fastqs` to a different location? I've observed that if `--runpath` and `--outdir` are the same, after I get the error above some of the reads have been deleted from the `fastqs` directory so I'm guessing this directory might be overwritten. I hope that makes sense!

Thanks again for your help.

**Comment by mandysulli** [comment]
The staging process does copy the fastqs folder to a new location. So if your runpath and outdir are set to the same location it will overwrite that fastqs folder with the copied fastqs folder. Though all of the sample should be there once the overwriting process is complete. How many samples are you working with, what is the size of the data and are you running this locally?

**Comment by gyebra-phs** [comment]
Hi Mandy, that's good to know thanks. I'm running the pipeline locally, and using a set of 16 samples (32 fastq files) with sizes from ~600Mb to ~2Gb).

**Comment by mandysulli** [comment]
Could you run the pipeline with just the 4 smallest samples and let me know if that works for you?

**Comment by gyebra-phs** [comment]
I tried that with no luck but then I ran the pipeline sucessfully with the example data provided on the github page so I went back to thinking there was an issue with the naming... And turns out if I change the read file names so that they all end in eg `R1.fastq.gz` (as opposed to `R1_001.fastq.gz` like in my sequences) it works! I thought the wildcards would have no issue but it was clearly upsetting some of the scripts.

I will keep testing the pipeline with different settings but I hope everything goes smoothly from now on! Sorry about all the fuss, but thanks a lot to you both for your help, it's really appreciated :)
